src/Utils/BlockTemplateUtils.php

```diff
@@ -198,6 +236,9 @@ public static function build_template_result_from_file( $template_file, $templat
 		$template->id      = $template_is_from_theme ? $theme_name . '//' . $template_file->slug : self::PLUGIN_SLUG . '//' . $template_file->slug;
 		$template->theme   = $template_is_from_theme ? $theme_name : self::PLUGIN_SLUG;
 		$template->content = self::inject_theme_attribute_in_content( $template_content );
+		if ( 'archive-product' === $template_file->slug ) {
```

Comment:
It would be great if we could leave a comment here just explaining why we're doing this.